python/taichi/lang/kernel_impl.py

```diff
@@ -226,6 +226,14 @@ def extract_arguments(self):
             if annotation is inspect.Parameter.empty:
                 if i == 0 and self.classfunc:
                     annotation = template()
+                # TODO: pyfunc also need type annotation check when real function is enabled,
+                #       but that has to happen at runtime when we know which scope it's called from.
+                elif not self.pyfunc and impl.get_runtime(
+                ).experimental_real_function:
+                    _taichi_skip_traceback = 1
+                    raise KernelDefError(
```

Comment:
Ah, how about we use `TaichiSyntaxError` for consistency?

Reply:
Are we going to keep both `TaichiSyntaxError` and `KernelDefError` in the future? Asking since all the surrounding code uses `KernelDefError` :P

Reply:
OK.. Then maybe we should change the missing return type to `KernelDefError` as well :-)

Reply:
haha let me show you this, you'll be amazed. ;) 
the missing return type is checked at runtime (where we use a lot  of `TaichiSyntaxError`): https://github.com/taichi-dev/taichi/blob/master/python/taichi/lang/stmt_builder.py#L631
while the missing annotation is checked at compile time (where we use a lot of `KernelDefError`) ;) https://github.com/taichi-dev/taichi/blob/master/python/taichi/lang/kernel_impl.py#L118 
For example, if you look at the tests I added below, you can trigger the error for missing input arg annotation by defining a ti.func. But triggering the error for missing return annotation have to be done by calling the ti.func :D

Reply:
We might need a bit discussion to decide which one (or two) errors to unify, maybe I just keep it locally consistent for now? ;)